lib/generate_cst.py

```diff
@@ -0,0 +1,162 @@
+
+def print_warning(msg):
+  print(msg, file=sys.stderr)
+
+def _install_dependency(name: str, dir_path: str) -> None:
+  print_warning(f"Checking {name} installation")
+  if not os.path.exists(os.path.join(dir_path, f"node_modules/{name}")):
+    print_warning(f"No {name} package found. Attempting `npm i {name}` in {dir_path}")
+    exit_installation = wrap_call(["npm", "i", name], cwd=dir_path)
+    if exit_installation != 0:
+      print_warning(f"Could not install tree-sitter. Try manually with `npm i {name}` inside {dir_path}")
+      sys.exit(exit_installation)
+
+def _check_and_install_tree_sitter(dir_path)-> None:
+  """
+    Checks tree-sitter-installation inside dir_path.
+    Ensures that tree-sitter and tree-sitter-cli are installed locally in node_modules
+  """
+  deps = ["nan", "tree-sitter", "tree-sitter-cli"]
+  for d in deps:
+    _install_dependency(d, dir_path)
+
+def _check_node_installation()-> None:
+  """
+    Checks node installation by invoking `node --version`
+  """
+  print_warning(f"Checking node installation")
+  exit_code = wrap_call(["node", "--version"])
+  if exit_code != 0:
+    print_warning(f"Please install NodeJS with `brew install node`")
+    sys.exit(exit_code)
+
+def _generate_json(grammar_dir: str) -> Optional[str]:
+  """
+    Generates grammar.json from grammar.js files of tree-sitter
+  """
+  print_warning(f"Generating grammar.json from {grammar_dir}")
+
+  # Sanity check with -V
+  subprocess.call([LOCAL_TREE_SITTER_PATH, "-V"], cwd=grammar_dir)
+
+  exit_code = subprocess.call([LOCAL_TREE_SITTER_PATH, "generate"], cwd=grammar_dir)
+  if exit_code != 0:
+    print("Could not generate grammar.json. Try running `tree-sitter generate` manually!")
+    sys.exit(1)
+  return os.path.join(grammar_dir, "grammar.json")
+
+def parse_grammar(grammar_path: str) -> Optional[Dict[str, Any]]:
+  """
+    Parses grammar.json in tree-sitter definitions. Generates
+    this file if necessary from grammar.js
+  """
+  _check_and_install_tree_sitter(grammar_path)
+  json_file = _generate_json(grammar_path)
+  with open(json_file, 'r') as generate_j:
+    try:
+      grammar = json.load(generate_j)
+    except JSONDecodeError as _:
+      print_warning(f"Invalid grammar.json file specified.")
+  if not grammar:
+    print_warning(f"Could not find grammar.json")
+    sys.exit(1)
+  return grammar
+
+def generate_cst_json_dumper(language_name, dir_path):
+  print_warning(f"Generating bindings for language '{language_name}' in {dir_path}")
+  template = f"""#!/usr/bin/env node
+
+const Parser = require('tree-sitter');
+const {language_name.capitalize()} = require('.');
+
+var args = process.argv.slice(2);
+
+var fs = require("fs");
+const sourceCode = fs.readFileSync(args[0]).toString();
+
+const parser = new Parser();
+parser.setLanguage({language_name.capitalize()});
+const tree = parser.parse(sourceCode);
+
+console.log(JSON.stringify(tree.rootNode, ["type", "children"], 2))
```

Comment:
can put a TODO here or a comment that later we will add startPOS, etc. but for now
we simplify things.